relay-sampling/src/lib.rs

```diff
@@ -366,46 +386,51 @@ impl TimeRange {
 #[serde(tag = "type")]
 pub enum DecayingFunction {
     #[serde(rename_all = "camelCase")]
-    Linear { decayed_sample_rate: f64 },
+    Linear { decayed_value: f64 },
     #[default]
     Constant,
 }
 
 /// A struct representing the evaluation context of a sample rate.
 #[derive(Debug, Clone, Copy)]
-enum SampleRateEvaluator {
+enum SamplingStrategyValueEvaluator {
     Linear {
         start: DateTime<Utc>,
         end: DateTime<Utc>,
-        sample_rate: f64,
-        decayed_sample_rate: f64,
+        initial_value: f64,
+        decayed_value: f64,
+        use_factor: bool,
     },
     Constant {
-        sample_rate: f64,
+        initial_value: f64,
     },
 }
 
-impl SampleRateEvaluator {
-    /// Evaluates the sample rate given the decaying function and the current time.
+impl SamplingStrategyValueEvaluator {
+    /// Evaluates the value of the sampling strategy given a the current time.
     fn evaluate(&self, now: DateTime<Utc>) -> f64 {
         match self {
-            SampleRateEvaluator::Linear {
+            SamplingStrategyValueEvaluator::Linear {
                 start,
                 end,
-                sample_rate,
-                decayed_sample_rate,
+                initial_value,
+                decayed_value,
+                use_factor,
             } => {
                 let now_timestamp = now.timestamp() as f64;
                 let start_timestamp = start.timestamp() as f64;
                 let end_timestamp = end.timestamp() as f64;
-                let progress_ratio = ((now_timestamp - start_timestamp)
-                    / (end_timestamp - start_timestamp))
-                    .clamp(0.0, 1.0);
+                let mut progress_ratio =
+                    (now_timestamp - start_timestamp) / (end_timestamp - start_timestamp);
+                if !use_factor {
+                    // In case we don't use factors, we want to clamp the value between 0 and 1.
+                    progress_ratio = progress_ratio.clamp(0.0, 1.0)
```

Comment:
The `progress_ratio` is computed entirely from timestamps, so why does it matter if the value is a factor or a sample rate < 1?

Reply:
I have just noticed that I have clamped the wrong field 🫣, I will need to clamp the result

Reply:
Will fix now, I also saw I forgot to remove the `true` for `use_factor`.